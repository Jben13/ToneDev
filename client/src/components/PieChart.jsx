import React from 'react';
import { Pie } from 'react-chartjs-2';
import PropTypes from 'prop-types';


const PieChart = (props) => {
  const { score } = props;
  const remainder = 100 - score;
  const data = {
    labels: ['Positive', 'Negative'],
    datasets: [
      {
        data: [score, remainder],
        backgroundColor: [
          '#2ec4b6',
          '#3b0066',
        ],
      },
    ],
  };

  const options = {
    title: {
      display: true,
      text: 'Sentinment Analysis Overview',
      fontFamily: 'Montserrat',
      fontSize: 20,
    },
  };

  return (
    <Pie data={data} options={options} />
  );
};
PieChart.propTypes = {
  score: PropTypes.number.isRequired,
};

export default PieChart;

// console.log(data.datasets[0].data[0]);
// console.log(data.datasets[0].data[1]);
// class PieChart extends React.Component {
//   constructor(props) {
//     super(props);
//     console.log(props);
//     this.state = {
//       data: {
//         labels: ['Positive', 'Negative'],
//         datasets: [
//           {
//             data: [75, 25],
//             backgroundColor: [
//               '#36A2EB',
//               '#FF6384',
//             ],
//           },
//         ],
//       },
//       options: {
//         title: {
//           display: true,
//           text: 'Sentinment Analysis Overview',
//           fontFamily: 'Roboto',
//           fontSize: 20,
//         },
//       },
//     };
//     this.setDisplay = this.setDisplay.bind(this);
//   }

//   componentDidMount() {
//     console.log(this.props);
//     this.setDisplay();
//   }

//   setDisplay() {
//     const { sentiment, score } = this.props;
//     console.log(sentiment);
//     console.log(score);
//     if (sentiment === 'positive') {
//       const remainder = 100 - score;
//       console.log(remainder);
//       this.setState({
//         data: {
//           labels: ['Positive', 'Negative'],
//           datasets: [
//             {
//               data: [score, remainder],
//               backgroundColor: [
//                 '#36A2EB',
//                 '#FF6384',
//               ],
//             },
//           ],
//         },
//       });
//     }
//   }

//   render() {
//     const { data, options } = this.state;
//     return (
//       <Pie data={data} options={options} />
//     );
//   }
// }
