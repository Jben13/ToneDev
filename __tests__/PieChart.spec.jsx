import React from 'react';
import PieChart from '../client/src/components/PieChart';

test('App renders component correctly', () => {
  const component = global.shallow(<PieChart />);
  expect(component).toMatchSnapshot();
});
