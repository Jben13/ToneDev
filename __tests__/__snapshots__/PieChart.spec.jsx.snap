// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component correctly 1`] = `
<Pie
  data={
    Object {
      "datasets": Array [
        Object {
          "backgroundColor": Array [
            "#36A2EB",
            "#FF6384",
          ],
          "data": Array [
            66,
            34,
          ],
        },
      ],
      "labels": Array [
        "Positive",
        "Negative",
      ],
    }
  }
  options={
    Object {
      "title": Object {
        "display": true,
        "fontFamily": "Roboto",
        "fontSize": 20,
        "text": "Sentinment Analysis Overview",
      },
    }
  }
/>
`;
